### PR TITLE
Rocfft issue

### PR DESCRIPTION
**Pull request by micmelesse**
This PR enable some failing unit tests for fft in pytorch.  

The reason these tests were failing was due to hipfft clobbering inputs causing a mismatch in tests that were checking that applying ffts and their reverse would get you back to the input.  

We solve this by cloning the input using an existing flag on the ROCM platform.

There PR doesnot enable all fft tests. There are other issues that need to be resolved before that can happen.

### CODE REVIEW

File: aten/src/ATen/native/cuda/CuFFTPlanCache.h [hunk 1]
```diff
@@ -255,7 +255,13 @@ class CuFFTConfig {
     // Since cuFFT has limited non-unit stride support and various constraints, we
     // use a flag to keep track throughout this function to see if we need to
     // input = input.clone();
+
+#ifdef __HIP_PLATFORM_HCC__
```
Comment (sunway513) [inline aten/src/ATen/native/cuda/CuFFTPlanCache.h:259]: @jeffdaily , do we still use this preprocessor? Or you'd recommend anything else for hipclang?

**Comment by micmelesse** [comment]
I am moving this PR to upstream master

File: aten/src/ATen/native/cuda/CuFFTPlanCache.h (hunk 1 above)
Comment (jeffdaily) [reply aten/src/ATen/native/cuda/CuFFTPlanCache.h:259]: still do, for now.  will be deprecated soon, but still the one to use.

**Comment by jeffdaily** [comment]
Moved here: https://github.com/pytorch/pytorch/pull/51581

**Pull request closed without merging**